internal/internal_task_handlers.go

```diff
@@ -508,6 +508,7 @@ func (w *workflowExecutionContextImpl) clearState() {
 	w.previousStartedEventID = 0
 	w.newCommands = nil
 	w.newMessages = nil
+	w.laTunnel = nil
```

Comment:
`clearState()`should be called if the worker is restarted and the workflow is in cache because the new worker will get the full history so it will consider the cached stale.